src/plugins/core/execute/processedHttpRegionInterceptor.ts

```diff
@@ -0,0 +1,69 @@
+import { HookInterceptor, HookTriggerContext } from 'hookpoint';
+
+import * as models from '../../../models';
+
+export class ProcessedHttpRegionInterceptor implements HookInterceptor<[models.ProcessorContext], boolean | void> {
+  id = 'processedHttpRegion';
+
+  public getResponseLoggingInterceptor(): HookInterceptor<
+    [models.HttpResponse, models.ProcessorContext],
+    boolean | void
+  > {
+    return {
+      id: this.id,
+      afterLoop: this.afterResponseLoggingLoop.bind(this),
+    };
+  }
+
+  public async beforeLoop(
+    hookContext: HookTriggerContext<[models.ProcessorContext], boolean | undefined>
+  ): Promise<boolean | undefined> {
+    const [context] = hookContext.args;
+    if (context.processedHttpRegions) {
+      const processedHttpRegion = this.toProcessedHttpRegion(context);
+      context.processedHttpRegions.push(processedHttpRegion);
+    }
+    return true;
+  }
+
+  public async afterLoop(
+    hookContext: HookTriggerContext<[models.ProcessorContext], boolean | undefined>
+  ): Promise<boolean | undefined> {
+    const [context] = hookContext.args;
+
+    const processedHttpRegion = context.processedHttpRegions?.find(obj => obj.id === context.httpRegion.id);
+    if (processedHttpRegion) {
+      processedHttpRegion.end = new Date().getTime();
+      processedHttpRegion.duration = processedHttpRegion.end - processedHttpRegion.start;
+      processedHttpRegion.disabled = !!hookContext.bail;
+      processedHttpRegion.metaData = {
+        ...context.httpRegion.metaData,
+      };
+      processedHttpRegion.testResults = context.httpRegion.testResults;
+    }
+    return true;
+  }
+
+  private async afterResponseLoggingLoop(
+    hookContext: HookTriggerContext<[models.HttpResponse, models.ProcessorContext], boolean | undefined>
+  ): Promise<boolean | undefined> {
+    const [response, context] = hookContext.args;
+
+    const processedHttpRegion = context.processedHttpRegions?.find(obj => obj.id === context.httpRegion.id);
+    if (processedHttpRegion) {
+      processedHttpRegion.request = response.request;
+      processedHttpRegion.response = response;
+    }
+    return true;
+  }
+
+  private toProcessedHttpRegion(context: models.ProcessorContext): models.ProcessedHttpRegion {
+    return {
+      id: context.httpRegion.id,
+      filename: context.httpFile.fileName,
+      symbol: context.httpRegion.symbol,
+      isGlobal: context.httpRegion.isGlobal(),
+      start: new Date().getTime(),
```

Comment:
purely for my own education, why opt for date time vs performance api? It looks like the performance api is more granular at the cost of being slower:

https://www.sitepoint.com/discovering-the-high-resolution-time-api/

Reply:
I actually wanted to change that back. Yes, I agree with you. `performance.now()` would be better, but I also wanted to output the current time and performance.now doesn't provide that. Ic didn't know the API and quickly changed it to Date. Now it is correct.